plugin/kotel/tracer.go

```diff
@@ -124,6 +124,7 @@ func (t *Tracer) WithProcessSpan(r *kgo.Record) (context.Context, trace.Span) {
 		semconv.MessagingDestinationKey.String(r.Topic),
 		semconv.MessagingOperationProcess,
 		semconv.MessagingKafkaPartitionKey.Int64(int64(r.Partition)),
+		semconv.MessagingKafkaMessageOffset.Int64(r.Offset),
```

Comment:
When is WithProcessSpan meant to be used? If this is while consuming, this is valid. If this is while producing, the Offset is not yet set -- it is only set when being unbuffered -- and this attribute will be invalid.

Reply:
With "WithProcessSpan" needs to be using in the consumption of records manually invoked by the user only for none batch processing, so during a fetch loop

Reply:
example:

```go
fetches.EachRecord(func(r *kgo.Record) {
   ctx, span :=  tracer.WithProcessSpan(r)
  ...
}
```

Reply:
Can it be used while producing?
```
r := kgo.StringRecord("foo")
ctx, span := tracer.WithProcessSpan(r)
cl.Produce(context.Background(), r, ...)
doSomethingWithSpan()
```
?

Reply:
It should not be from what I understand, as the readme does not show this. I can update the function comment to say to only use during consumption?